From a750247608b6a5f8231317132c22e15b5854bcc8 Mon Sep 17 00:00:00 2001
From: Ming Zhou <b42586@freescale.com>
Date: Wed, 3 Apr 2013 09:18:15 +0800
Subject: [PATCH 3/6] ENGR00242389:Video can't play if set the trim point then
 replay the video. 100%

remove some code to avoid looping endlessly when trim video.

Signed-off-by: Ming Zhou <b42586@freescale.com>
---
 src/com/android/gallery3d/app/TrimVideo.java | 22 ++++++++++++++--------
 1 file changed, 14 insertions(+), 8 deletions(-)

diff --git a/src/com/android/gallery3d/app/TrimVideo.java b/src/com/android/gallery3d/app/TrimVideo.java
index b0ed8e6..17534f1 100644
--- a/src/com/android/gallery3d/app/TrimVideo.java
+++ b/src/com/android/gallery3d/app/TrimVideo.java
@@ -175,19 +175,25 @@ public class TrimVideo extends Activity implements
     // to be updated immediately.
     private int setProgress() {
         mVideoPosition = mVideoView.getCurrentPosition();
+
+        //seek position is not accurate since video needs to start playback from key frame
+        //the actual position after seek may be earlier or later than the wanted position
+        //so remove these code to avoid looping endlessly
+
         // If the video position is smaller than the starting point of trimming,
         // correct it.
-        if (mVideoPosition < mTrimStartTime) {
-            mVideoView.seekTo(mTrimStartTime);
-            mVideoPosition = mTrimStartTime;
-        }
+        //if (mVideoPosition < mTrimStartTime) {
+        //    mVideoView.seekTo(mTrimStartTime);
+        //    mVideoPosition = mTrimStartTime;
+        //}
+
         // If the position is bigger than the end point of trimming, show the
         // replay button and pause.
         if (mVideoPosition >= mTrimEndTime && mTrimEndTime > 0) {
-            if (mVideoPosition > mTrimEndTime) {
-                mVideoView.seekTo(mTrimEndTime);
-                mVideoPosition = mTrimEndTime;
-            }
+           //if (mVideoPosition > mTrimEndTime) {
+           //    mVideoView.seekTo(mTrimEndTime);
+           //    mVideoPosition = mTrimEndTime;
+           //}
             mController.showEnded();
             mVideoView.pause();
         }
-- 
1.8.0

